src/server.ts

```diff
@@ -243,11 +358,32 @@ export class Server extends EventEmitter {
         // We need to consume socket errors, because the handlers are attached asynchronously.
         // See https://github.com/apify/proxy-chain/issues/53
         socket.on('error', (err) => {
-            // Handle errors only if there's no other handler
-            if (this.listenerCount('error') === 1) {
+            // Prevent duplicate error handling for the same socket
+            if (socket.proxyChainErrorHandled) return;
+            socket.proxyChainErrorHandled = true;
+
+            // Log errors only if there are no user-provided error handlers
+            if (this.listenerCount('error') === 0) {
                 this.log(socket.proxyChainId, `Source socket emitted error: ${err.stack || err}`);
             }
         });
+
+        // Check once per connection for socket._parent availability.
+        if (this.serverType === 'https') {
+            const rawSocket = socket._parent;
```

Comment:
It may be worth asking in https://github.com/nodejs/node about the safety of using this private property.